### DISCUSSION TITLE
Onvif stiill?

### DISCUSSION THREAD

**Discussion by guarddog13** [gtxaspec/wz_mini_hacks#284] [Development]
Is there still plans to make these Onvif capable?

**Comment by KevinKadow** [comment]
Shouldn't be a huge lift to extend the webserver to appear as ONVIF.

If the client supports manually adding a camera, it will need gSOAP support and respond to the minimal set of queries (particularly GetDeviceInformation, GetCapabilities   GetServices and GetStreamUri).   See also https://github.com/onvif/testspecs and https://github.com/KoynovStas/onvif_srvd

Discovery is trickier, needs a multicast listener to be discoverable via Web Services Dynamic Discovery.

I would be happy to just have a basic V3 camera manually added and streamed; adding PTZ and forwarding in-camera motion events would be a stretch goal.

**Comment by guarddog13** [comment]
All I want is motion events forwarded to AgentDVR so I can run deepstack beside it without much cpu usage. Using the scripts is nice but they timeout about once a week and with 7 cams that's one reboot a day. Plus the scripts are only so reliable. If motion stops right after the last people detect call to deepstack the sensor in HA gets stuck on.
